ODLA/platforms/vodla/vODLA.cpp

```diff
@@ -524,21 +498,36 @@ odla_status odla_BindToOutput(odla_value value, odla_void* data_ptr,
 }
 
 odla_status odla_CreateContext(odla_context* context) {
-  *context = new _odla_context();
+  std::cout << "odla_CreateContext handler " << g_dev->vodh_hd
+            << " dev_list ptr " << g_dev->vodh_dev_list << " ctx " << context
+            << ", *ctx " << *context << std::endl;
 
-  if (context == NULL) {
-    std::cout << "[vODLA] ERROR: failed to create odla context.\n";
+  vodh_ret ret =
+      vodh_create_context(g_dev->vodh_hd, g_dev->vodh_dev_list, context);
+
+  if (context == NULL || ret) {
+    std::cout << "[vODLA] ERROR: failed to create odla context. context "
+              << context << " ret " << ret << "\n";
     return ODLA_FAILURE;
   }
 
-  (*context)->Id = g_ctxId++;
+  {
+    std::lock_guard<std::mutex> lck(cnt_mu_);
```

Comment:
maybe easier if you define g_ctxId as std::atomic<int>, then you just do (*context)->Id = g_ctxId++ as before

Reply:
solved